test/PensionFundReleaseTest.js

```diff
@@ -1,87 +1,178 @@
+      await fund.vote(false, "justification", {from: VALIDATORS[1]})
+      let approval = await fund.isBurnApproved.call()
+      approval.should.be.equal(true)
+    })
+
+    it("#8 should not allow burn if not all VALIDATORS rejected release", async () => {
+      await fund.vote(false, "justification", {from: VALIDATORS[0]})
+      let approval = await fund.isBurnApproved.call()
+      approval.should.be.equal(false)
+    })
+
+    it("#9 should release roots if all conditions met", async () => {
+      let workerBalance, balanceNumber
+      let balance = INITIAL_BALANCE      
+      await token.transfer(fund.address, balance)
+      await fund.vote(true, "justification", {from: VALIDATORS[0]})
+      await fund.vote(true, "justification", {from: VALIDATORS[1]})
+      let release = await fund.releaseRoots({from: WORKER})
+      release.logs[0].args.amount.toNumber().should.be.equal(balance* 20 / 100)
+      workerBalance = await token.balanceOf(WORKER)
+      workerBalance.toNumber().should.be.equal(balance * 20 / 100)
+    })
+
+    it("#10 should attempt to release roots and fail", async () => {
+      let balance = INITIAL_BALANCE
+      fund = await PensionFundRelease.new.apply(
+        this,
+        deployParams(
+          firstPaymentTime + TIME_INCREMENT,
+          FIRST_PAYMENT_PERCENT,
+          token.address,
+          PAYOUT_PERCENT
+        )
+      )         
+      await token.transfer(fund.address, balance)
+      await fund.vote(true, "justification", {from: VALIDATORS[0]})
+      await fund.vote(true, "justification", {from: VALIDATORS[1]})
+      try {
+        await fund.releaseRoots({from: WORKER})
+      } catch (err) {
+        err.should.be.a("Error")
+      }
+    })
+
+    it("#11 should release all roots over time", async () => {
+      let workerBalance, balanceNumber
+      let balance = INITIAL_BALANCE      
+      // Transfer previous balance away
+      await token.transfer(UNAUTHORIZED, 20, {from: WORKER})
+      await token.transfer(fund.address, balance)
+      await fund.vote(true, "justification", {from: VALIDATORS[0]})
+      await fund.vote(true, "justification", {from: VALIDATORS[1]}) 
+      let release = await fund.releaseRoots({from: WORKER})
+      while(await fund.balance() > 0){
+        web3.currentProvider.send({
+          jsonrpc: '2.0',
+          method: 'evm_increaseTime',
+          params: [TIME_INCREMENT], 
+          id: new Date().getTime() 
+        })
+        release = await fund.releaseRoots({from: WORKER})
+        release.logs[0].args.amount.toNumber().should.be.equal(40)             
+      }
+      workerBalance = await token.balanceOf(WORKER)
+      workerBalance.toNumber().should.be.equal(INITIAL_BALANCE)
+      try {
+        await fund.releaseRoots({from: WORKER})
+      } catch (err) {
+        err.should.be.a("Error")
+      }    
+    })
+
+    it("#12 should release all roots and then release", async () => {
```

Comment:
It was meant to test that a release would fail after all funds are released, maybe rename ``released all roots and then fail to release further ``?